docs/triggers.md

```diff
@@ -51,6 +51,28 @@ spec:
     ref: pipeline-template
 ```
 
+Triggers can reference templates and bindings based on data in the event body, headers, or processed extensions. This allows contextual information to be pulled in from the event to determine which binding or template should be populated. The `ref` field for `bindings` and `templates` can be populated in the same manner as triggertemplate params.
+
+This can be useful when different pipelines require different resources or workspaces to run. `TriggerTemplate` objects by themselves must have a static configuration structurually, even if individual strings are able to be populated.
+
+<!-- FILE: examples/triggers/trigger-with-event-reference.yaml -->
+```YAML
+apiVersion: triggers.tekton.dev/v1alpha1
+kind: Trigger
+metadata:
+  name: trigger
+spec:
+  interceptors:
+    - cel:
+        overlays:
+        - key: extensions.truncated_sha
+          expression: "body.pull_request.head.sha.truncate(7)"
+  bindings:
+  - ref: pipeline-$(body.repository.owner.login)-binding
```

Comment:
what happens if the expression does not resolve? (Or it resolves to some other binding that was unintended.

Reply:
I'm not sure what `unintended` means in this context. 

Failure scenarios operate in much the same way as other tekton expressions within a Trigger:

1. If the expression is not found, processing is stopped with an error message saying that the key could not be found in the event
2. If the triggerbinding/triggertemplate is not found, processing is stopped as it retrieves a 404 from the getTB/getCTB/getTT function in the same was as if a non-existent triggertemplate was referenced.

Reply:
would explicitly specifying a default(s) as additional field(s) in our new API if 1) and/or 2) occur provide any value?

apologies, but I don't remember from the conversation I was a part of at the community meeting a couple of weeks ago got into any order of precedence / defaulting wrt the existing API in this space

Reply:
one way defaulting could work is by setting both a `ref` and `dynamicRef` fields. During processing, the dynamic reference could take first priority, with the `ref` being a fallback if the dynamic processing fails.

That might be challenging from a readability perspective, so maybe an additional field would be better for specifying this default value when dynamic resolution fails.

Reply:
Yeah might need some order of precedence defined for the `ref` and `dynamicRef` regardless, unless we updated the client side validation and/or add a webhook to prevent both being set.  But minimally some updates to the markdown docs for these api explaining what it would be.

But I tend to side on your thought on the additional field being better.  Plus IIRC there is some precedence in other portion(s) of the Trigger API for default values when resolving stuff of this ilk.

Reply:
@gabemontero kind of dropped off this conversation thread. I think `dynamicRef` takes precedence over `ref` if present. I'll update this PR to use the additional field tomorrow and you can let me know what you think about this PR after the changes

Reply:
sounds good @jmcshane